internal/provider/auth.go

```diff
@@ -80,19 +81,57 @@ func (l *AuthLoginCommon) Method() string {
 	return ""
 }
 
-func (l *AuthLoginCommon) copyParams(excludes ...string) map[string]interface{} {
+func (l *AuthLoginCommon) copyParams(includes ...string) (map[string]interface{}, error) {
 	params := make(map[string]interface{}, len(l.params))
-	for k, v := range l.params {
-		params[k] = v
+	if len(includes) == 0 {
+		for k, v := range l.params {
+			params[k] = v
+		}
+	} else {
+		var missing []string
+		for _, k := range includes {
+			v, ok := l.params[k]
+			if !ok {
+				missing = append(missing, k)
+				continue
+			}
+			params[k] = v
+		}
+		if len(missing) > 0 {
+			return nil, fmt.Errorf("missing params %v", missing)
+		}
 	}
 
+	return params, nil
+}
+
+func (l *AuthLoginCommon) copyParamsExcluding(excludes ...string) map[string]interface{} {
+	params, _ := l.copyParams()
 	for _, k := range excludes {
 		delete(params, k)
 	}
 
 	return params
 }
 
+func (l *AuthLoginCommon) copyParamsIncluding(includes ...string) (map[string]interface{}, error) {
+	missing := []string{}
+	params := map[string]interface{}{}
+	for _, k := range includes {
```

Comment:
nit: some duplicated code in here and `copyParams`; wondering if it's worth refactoring

Reply:
Fixed in 2855a9dc194b4b3cf2caf0935b8997265b78f979